src/canisters/frontend/ic-certified-assets/src/tests.rs

```diff
@@ -232,6 +240,67 @@ fn can_create_assets_using_batch_api() {
     );
 }
 
+#[test]
+fn serve_correct_encoding() {
+    let mut state = State::default();
+    let time_now = 100_000_000_000;
+
+    const IDENTITY_BODY: &[u8] = b"<!DOCTYPE html><html></html>";
+    const GZIP_BODY: &[u8] = b"this is 'gzipped' content";
+
+    create_assets(
+        &mut state,
+        time_now,
+        vec![
+            AssetBuilder::new("/contents.html", "text/html")
+                .with_encoding("identity", vec![IDENTITY_BODY]),
+            AssetBuilder::new("/contents.html", "text/html").with_encoding("gzip", vec![GZIP_BODY]),
+        ],
+    );
```

Comment:
It would also be good to also show what happens if there is only an `identity` encoding (ic-asset only stores a gzip encoding if there is a size reduction), and only `gzip` is requested.  The identity encoding should be returned in this case.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Encoding